src/components/select-project/select-project.controller.ts

```diff
@@ -0,0 +1,93 @@
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+export class SelectProjectController implements angular.IController {
+  static $inject = ['$scope', '$filter', 'DataService', 'ProjectsService', 'Logger', 'AuthService'];
+
+  public ctrl: any = this;
+  public $scope: any;
+
+  private $filter: any;
+  private DataService: any;
+  private ProjectsService: any;
+  private Logger: any;
+  private AuthService: any;
+
+  constructor($scope: any, $filter: any, DataService: any, ProjectsService: any, Logger: any, AuthService: any) {
+    this.$scope = $scope;
+    this.$filter = $filter;
+    this.DataService = DataService;
+    this.AuthService = AuthService;
+    this.ProjectsService = ProjectsService;
+    this.Logger = Logger;
+  }
+
+  public $onInit() {
+    this.ctrl.nameTaken = false;
+    // TODO disableInputs?
+    this.ctrl.disableInputs = false;
+
+    this.ProjectsService.canCreate().then(() => {
+      this.ctrl.canCreate = true;
+      this.AuthService.withUser().then(this.onVerifyUser);
+    }, (result) => {
+      this.ctrl.canCreate = false;
+      // 403 Forbidden indicates the user doesn't have authority.
+      // Any other failure status is an unexpected error.
+      if (result.status !== 403) {
+        var msg = 'Failed to determine create project permission';
+        if (result.status !== 0) {
+          msg += " (" + result.status + ")";
+        }
+        this.Logger.warn(msg);
+      }
+      this.AuthService.withUser().then(this.onVerifyUser);
+    });
+  }
+
+  public $onChanges(onChangesObj: angular.IOnChangesObject) {
+    if (onChangesObj.nameTaken && !onChangesObj.nameTaken.isFirstChange()) {
+      this.ctrl.nameTaken = onChangesObj.nameTaken.currentValue;
+      if (this.ctrl.nameTaken) {
+        this.ctrl.selectedProject.validName = false;
+      }
+    }
+  }
+
+  public onVerifyUser = () => {
+    this.listProjects();
+  };
+
+  public onNewProjectNameChange() {
+    this.ctrl.nameTaken = false;
+    this.ctrl.selectedProject.validName = this.ctrl.createProjectForm.$valid;
+  }
+
+  public isNewProject(): boolean {
+    return this.ctrl.selectedProject && this.ctrl.selectedProject.metadata ? this.ctrl.selectedProject.metadata.uid === 'create_project' : false;
+  }
+
+  private listProjects() {
+    this.DataService.list('projects', this.$scope).then((response: any) => {
+      // 'Create Project' placeholder obj in the dropdown
+      let createProject = {
+        "metadata": {
+          "uid": "create_project",
+          "annotations": {
+            "openshift.io/display-name": "Create Project",
+            "new-display-name": ""
+          },
+          validName: false
+        }
+      };
+      this.ctrl.projects = _.sortBy(response.by('metadata.name'), this.$filter('displayName'));
+
+      // get most recently created
+      this.ctrl.selectedProject = _.first(this.$filter('orderObjectsByDate')(this.ctrl.projects, true));
```

Comment:
You can loop through the projects once -- O(n) -- to find most recent rather than sorting -- O(n log(n)). It's possible for admins to see hundreds or thousands of projects, so it can make a difference.

Here is an example:

https://github.com/openshift/origin-web-console/blob/8f4cda159d1382b7e5e2d4c3223b4ac248be6e1f/app/scripts/filters/date.js#L99-L110